allure/allure.py

```diff
@@ -45,37 +49,90 @@ def before_scenario(self, scenario_name):
         etree.SubElement(labels, "label", attrib={"name": "host", "value": "autotester"})
         self._steps = etree.SubElement(self._test_case, "steps")
 
+        self.scenario_exception = None
+        self.scenario_steps = scenario.steps
+
     def after_scenario(self, status):
         """Set scenario status and end time
 
         :param status: scenario execution status
         """
-        self._test_case.attrib["status"] = status
+        try:
+            stat = status.split('.')[1]
+        except IndexError:
+            stat = 'Unknown'
+
+        if not self._test_case.attrib.get("status", None):
+            self._test_case.attrib["status"] = stat
         self._test_case.attrib["stop"] = get_time()
 
+        if self.scenario_exception:
+            scen_failure = etree.SubElement(self._test_case, "failure")
+            scen_failure_msg = etree.SubElement(scen_failure, "message")
+            format_exception = str(type(self.scenario_exception)).split(' ')[1][1:-2] + ':'
+            scen_failure_msg.text = '{} {}'.format(format_exception, self.scenario_exception)
+            scen_failure_stack_trace = etree.SubElement(scen_failure, "stack-trace")
+
     def before_step(self, step_name):
         """Create step
 
         :param step_name: step name
         """
-        self._step = etree.SubElement(self._steps, "step", attrib={"start": get_time()})
-        name = etree.SubElement(self._step, "name")
-        name.text = step_name
-        title = etree.SubElement(self._step, "title")
-        title.text = step_name
-        etree.SubElement(self._step, "attachments")
-        etree.SubElement(self._step, "steps")
-
-    def after_step(self, status):
+        try:
```

Comment:
please use 
`if hasattr(a, 'property'):
     a.property
`
https://stackoverflow.com/questions/610883/how-to-know-if-an-object-has-an-attribute-in-python